### DISCUSSION TITLE
Why was the license changed from MPL 2.0 to GPL 3.0 in September 2022?

### DISCUSSION THREAD

**Discussion by fsiek** [FactoryXCode/MfPack#43] [Q&A] [answered]
So, it seams you no longer want to have your API translation to be used in commercial applications.
Are all contributors happy with that decision?
It creates a difficult license situation as older releases are still under MPL 2.0 and Ciaran3's or adaloveless forks for example are MPL 2.0, too. Does anyone know if all forks have to change their licenses, too? Or can the forks have their own license?

Is it even allowed to change MPL code to another license?

In the Mozilla 2.0 FAQs you can read under Q11: _The license requires that Modifications (as defined in Section 1.10 of the license) must be licensed under the MPL and made available to anyone to whom you distribute the Source Code._ 
Link: [https://www.mozilla.org/en-US/MPL/2.0/FAQ/](https://www.mozilla.org/en-US/MPL/2.0/FAQ/)

FS

**Comment by FactoryXCode** [comment]
Hello FS,

The reason of changing the license is that the license used on MfPack versions lower than version 3.1.3 are outdated (But will be keeping their license conditions). At the time it served well, but did not cover all  issues covering the MfPack translated API's to be included in commercial distributions. Like, for example the translated API's DirectShow or the Clootie translations of DirectX included by Embarcadero's Delphi compilers.

To be clear: You are allowed to use the MfPack translated API's for commercial use to write your software, but you are not allowed to include, distribute or sell the MfPack translated API's as part of your software for commercial use.
Licenses can be changed for every MfPack version or distribution when required. 
The header of each file containing the legal stuff is valid for those translations.

So,
1st: Yes we think all contributors are happy with that.
2th: We are not responsible for forks not following changes from the master sources.

Regards, Tony.

**Reply by hotrs** [reply]
I need some more information about this topic.

Using the MfPack under the GPL v3 license in a program means that the program is "derived work" and must be licensed under the GPL v3, too. And therefore the source of the program must be make available. A point which most commercial programs can't achieve.

I'am right here?

**Reply by fsiek** [reply]
Yes, that's right and that exactly is the problem with the change to GPL. GPL is viral and if you use GPL code, you have to give away your source code, too.

**Reply by FactoryXCode** [reply]
GPL 3.0 is a Copyleft license, so was the former license.

A short interpretation: 
The freedom to use the work, the freedom to study the work, the freedom to copy and share the work with others,
the freedom to modify the work, and the freedom to distribute modified and therefore derivative works.

The copyleft license is superseded under chapter Explanatory memorandum, as part of the header.
The most important claim is that commercial use, forbids the distribution of this sourcecode.
So, that includes "derived work" too.

Also, the definition of a derived work concerning sourcecode is, for example, plunging the whole sourcecode into one MfPack .pas  file or something like that. Using sourcecode to compile to a binary has - in my opinion - nothing to do with derivative works. Unless you want to distribute the sourcecode as a part of your product... And that is prohibited...

**Comment by FactoryXCode** [comment, marked as answer]
Good news: 
With the introduction of MfPack 3.1.4 the license has been changed to MPL version 2.0 to match the MIT licenses from Microsoft.
